### DISCUSSION TITLE
Challenges and Considerations for Implementing Opentcs in RGB Vehicle Scenarios for Intelligent Warehousing

### DISCUSSION THREAD

**Discussion by ericbin** [openTCS/opentcs#79] [Q&A] [answered]
Opentcs is a great project, and I plan to use it in a RGV vehicle scenario for intelligent warehousing. The RGV vehicles run on rail-mounted shelves, which are composed of rectangular small grids about 2 meters long. Opentcs sends information to the vehicles in the form of one move command at a time. In my scenario, there is a situation where the vehicle stops and starts again due to the short distance between two positioning points.

How can I handle this situation to enable the vehicles to receive multiple move commands at once, avoid collisions when multiple vehicles are present, and prevent locking up long routes, which could cause other vehicles to be unable to run for a long time?
![This is an image](https://pic2.zhimg.com/v2-4efdb505a438f099f6017f1dec14d751_r.jpg)

**Comment by swltr** [comment, marked as answer]
Your vehicle driver can get multiple/more commands in advance by setting `sentQueueCapacity` accordingly - [see JavaDoc](https://www.opentcs.org/docs/5.7/developer/api-base/org/opentcs/drivers/vehicle/BasicVehicleCommAdapter.html#%3Cinit%3E(org.opentcs.drivers.vehicle.VehicleProcessModel,int,int,java.lang.String,java.util.concurrent.ScheduledExecutorService)). (Also set `commandQueueCapacity` to `sentQueueCapacity + 1`.)

**Reply by ericbin** [reply]
thanks a lot
